src/scripts/main.js

```diff
@@ -1,3 +1,57 @@
 'use strict';
 
-// write code here
+const appendRow = document.querySelector('.append-row');
+const removeRow = document.querySelector('.remove-row');
+const appendColumn = document.querySelector('.append-column');
+const removeColumn = document.querySelector('.remove-column');
+const table = document.querySelector('tbody');
+const maxItems = 10;
+const minItems = 2;
+
+appendRow.addEventListener('click', () => {
+  const clone = table.rows[0].cloneNode(true);
+
+  table.append(clone);
+
+  if (table.rows.length === maxItems) {
+    appendRow.setAttribute('disabled', 'true');
+  }
+
+  removeRow.removeAttribute('disabled');
+});
+
+removeRow.addEventListener('click', () => {
+  table.lastElementChild.remove();
+
+  if (table.rows.length === minItems) {
+    removeRow.setAttribute('disabled', 'true');
+  }
+
+  appendRow.removeAttribute('disabled');
+});
+
+appendColumn.addEventListener('click', () => {
+  [...table.children].map((row) => {
```

Comment:
Do you really need the map? what about forEach?
And as I see you are not mutating initial array, only manipulate with DOM, so I suggest to not copy it
```suggestion
  table.children.forEach((row) => {
```